src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,132 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double RESIZE_COEFFICIENT = 1.5;
+    private Object[] array;
+    private Object[] tempArray;
+    private int size = 0;
+
+    public ArrayList(int capacity) {
+        if (capacity > 0) {
+            this.array = new Object[capacity];
+        } else {
+            this.array = new Object[DEFAULT_CAPACITY];
+        }
+    }
+
+    public ArrayList() {
+        this.array = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (this.size + 1 > array.length) {
+            resizeArray(array);
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index " + index);
+        } else if (index == size) {
+            this.add(value);
+        } else {
```

Comment:
```suggestion
        if (index == size) {
            add(value);
        }
        checkIndex(index);
```

Reply:
I rewritten this part with your variant.. In my opinion it's become more complicated to read it, because you need to remember all 'if' cases.